iceoryx_posh/include/iceoryx_posh/internal/roudi/iceoryx_port_pool.hpp

```diff
@@ -28,8 +28,8 @@ class IceOryxPortPool : public PortPool
   public:
     IceOryxPortPool(PortPoolData& portPool) noexcept;
 
-    cxx::vector<SenderPortType::MemberType_t*, MAX_PORT_NUMBER> senderPortDataList() noexcept override;
-    cxx::vector<ReceiverPortType::MemberType_t*, MAX_PORT_NUMBER> receiverPortDataList() noexcept override;
+    cxx::list<SenderPortType::MemberType_t, MAX_PORT_NUMBER>& senderPortDataList() noexcept override;
```

Comment:
Uh this is dangerous. What if someone takes the list and modifies it arbitrarily? Is this what we want? If yes, then we can just make the whole list public, otherwise I would return here a copy.

Reply:
A discussion lead to the following proposal: return a container (probably still a vector if it is just read from) of pointers to the ports as is instead of a reference to the underlying container, which is now a list. 
This more overhead since this container needs to be constructed everytime the function is called (this is also the case now), but does not expose the internal list (but the ports, as before).

In the long run this can maybe be improved even more by passing the list reference as proposed, at the cost of losing encapsulation.

The underlying container should still be changed to a list to allow efficient insertion and removal.